### DISCUSSION TITLE
Any option to make nvim-tree window keep current window width when opening another file from nvim-tree?

### DISCUSSION THREAD

**Discussion by ljie-PI** [nvim-tree/nvim-tree.lua#2915] [Q&A] [answered]
Sometimes name of files are long and I change the width of the nvim-tree window to view the full names. Then I may open that file from nvim-tree. The nvim-tree window gets reset to pre-defined width after I open the file, and I need to resize the nvim-tree window again to see the full name of files.

Is there an option that can let nvim-tree window keep the width?

**Comment by alex-courtis** [comment, marked as answer]
You have a lot of options for width, including a function to define whatever behaviour you want. See `:help nvim-tree.view.width`

You could try dynamic width: set a min and max to suit.

I have a mapping `A` that toggles between fixed width 30 and unlimited (adaptive) width: https://github.com/alex-courtis/arch/blob/master/config/nvim/lua/amc/plugins/nvt.lua#L103

**Reply by ljie-PI** [reply]
Thanks @alex-courtis . I see your are switching between VIEW_WIDTH_FIXED and -1. Does "-1" means current width? I tried directly using -1, but seems not working as expected.

**Reply by alex-courtis** [reply]
`-1` indicates unbounded - fit to the longest name.

I usually use it, however must turn it off for deep codebases, such as java.

If you're having trouble getting functions etc. going, you could try hardcoding to demonstrate the functionality:
```lua
  view = {
    width = {
      max = -1,
    },
  },
```

**Reply by ljie-PI** [reply]
Hi @alex-courtis , I saw there is "preserve_window_proportions". Is this intended to be used to keep current size and don't resize the windows?

**Reply by alex-courtis** [reply]
Yes. That will prevent a `:wincmd =` when sizing the window.
